### PR TITLE
EN 882 list data types

### PR DESCRIPTION
**Pull request by mcarta-rt**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: sass/common/table/TableList.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.TableList {
```
Comment (alessioscalici) [inline sass/common/table/TableList.scss]: Please move these styles to the specific component, following BEM

File: src/api/dataTypes.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
-  new Promise((resolve, reject) => {
-    if (type.errors) {
-      throttle(() => resolve(DATA_TYPES.payload));
-    } else {
-      reject(ERROR);
-    }
-  })
-);
+export const getDataTypes = (page = 1, params) => new Promise((resolve) => {
+  if (params) {
```
Comment (alessioscalici) [inline src/api/dataTypes.js]: Maybe throttling could be useful here. @nicpuddu thoughts?

File: src/app-init/router.js [hunk 3]
```diff
@@ -46,6 +48,8 @@ routerConfig(
       { name: ROUTE_FRAGMENT_EDIT, path: '/fragment/edit/:fragmentCode' },
       { name: ROUTE_FRAGMENT_DETAIL, path: '/fragment/view/:fragmentCode' },
       { name: ROUTE_DATA_MODEL_ADD, path: '/datamodel/add' },
+      { name: ROUTE_DATA_TYPE_LIST, path: '/datatype' },
+      { name: ROUTE_DATA_TYPE_EDIT, path: '/datatype/edit/:datatype' },
```
Comment (alessioscalici) [inline src/app-init/router.js]: It would be more coherent to name the param dataTypeCode

File: src/ui/data-types/list/DataTypeListMenuActions.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  render() {
+    const { onClickDelete } = this.props;
+    const editLabel = (
+      <FormattedMessage id="app.edit" values={{ code: this.props.code }} />
+    );
+    return (
+      <DropdownKebab pullRight id={`${this.props.code}-actions`}>
+        {/* <LinkMenuItem
```
Comment (alessioscalici) [inline src/ui/data-types/list/DataTypeListMenuActions.js]: Please fix comments

File: src/ui/data-types/list/DataTypeListTable.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    if (this.props.datatypes.length > 0) {
+      const pagination = {
+        page: this.props.page,
+        perPage: this.props.pageSize,
+        perPageOptions: [5, 10, 15, 25, 50],
+      };
+
+      return (
+        <Col md={12}>
+          <table className="DataTypeListTable__table TableList table table-striped table-bordered">
```
Comment (alessioscalici) [inline src/ui/data-types/list/DataTypeListTable.js]: Move TableList styles into BEM DataTypeListTable__table class

File: src/ui/data-types/list/DataTypeListTable.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            <thead>
+              <tr>
+                <th><FormattedMessage id="app.name" /></th>
+                <th className="TableListRow__th-sm text-center">
```
Comment (alessioscalici) [inline src/ui/data-types/list/DataTypeListTable.js]: Here too

File: test/ui/data-types/list/DataTypeListMenuActions.test.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+describe('DataTypeListMenuActions', () => {
+  let component;
+  beforeEach(() => {
+    component = shallow(<DataTypeListMenuActions code="code" />);
+  });
+
+  it('errors without a code', () => {
+    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+    shallow(<DataTypeListMenuActions />);
+    expect(consoleError).toHaveBeenCalledWith('Warning: Failed prop type: The prop `code` is marked as required in `DataTypeListMenuActions`, but its value is `undefined`.\n    in DataTypeListMenuActions (at DataTypeListMenuActions.test.js:19)');
```
Comment (alessioscalici) [inline test/ui/data-types/list/DataTypeListMenuActions.test.js]: Very long line

File: test/ui/data-types/list/DataTypeListTable.test.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  });
+
+  it('renders without crashing', () => {
+    expect(component.exists()).toEqual(true);
+  });
+
+  it('errors without a page', () => {
+    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+    shallow(<DataTypeListTable pageSize={1} totalItems={1} />);
+    expect(consoleError).toHaveBeenCalledWith('Warning: Failed prop type: The prop `page` is marked as required in `DataTypeListTable`, but its value is `undefined`.\n    in DataTypeListTable (at DataTypeListTable.test.js:27)');
```
Comment (alessioscalici) [inline test/ui/data-types/list/DataTypeListTable.test.js]: No need to check message

File: test/ui/data-types/list/DataTypeListTable.test.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    consoleError.mockReset();
+    consoleError.mockRestore();
+  });
+
+  it('errors without a pageSize', () => {
+    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+    shallow(<DataTypeListTable page={1} totalItems={1} />);
+    expect(consoleError).toHaveBeenCalledWith('Warning: Failed prop type: The prop `pageSize` is marked as required in `DataTypeListTable`, but its value is `undefined`.\n    in DataTypeListTable (at DataTypeListTable.test.js:35)');
```
Comment (alessioscalici) [inline test/ui/data-types/list/DataTypeListTable.test.js]: No need to check message

File: test/ui/data-types/list/DataTypeListTable.test.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    consoleError.mockReset();
+    consoleError.mockRestore();
+  });
+
+  it('errors without totalItems', () => {
+    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+    shallow(<DataTypeListTable pageSize={1} page={1} />);
+    expect(consoleError).toHaveBeenCalledWith('Warning: Failed prop type: The prop `totalItems` is marked as required in `DataTypeListTable`, but its value is `undefined`.\n    in DataTypeListTable (at DataTypeListTable.test.js:43)');
```
Comment (alessioscalici) [inline test/ui/data-types/list/DataTypeListTable.test.js]: No need to check message

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Pull request merged**